pkg/controller/clusterpool/clusterpool_controller_test.go

```diff
@@ -108,6 +119,39 @@ func TestReconcileClusterPool(t *testing.T) {
 			},
 			expectedMissingDependenciesStatus: corev1.ConditionUnknown,
 			expectedCapacityStatus:            corev1.ConditionUnknown,
+			expectedCDCurrentStatus:           corev1.ConditionUnknown,
+		},
+		{
+			name: "poolVersion changes with Platform",
+			existing: []runtime.Object{
+				initializedPoolBuilder.Build(testcp.WithPlatform(hivev1.Platform{
+					AWS: &aws.Platform{
+						CredentialsSecretRef: corev1.LocalObjectReference{Name: "foo"},
+					},
+				})),
+			},
+			expectPoolVersionChanged: true,
+		},
+		{
+			name: "poolVersion changes with BaseDomain",
+			existing: []runtime.Object{
+				initializedPoolBuilder.Build(testcp.WithBaseDomain("foo.example.com")),
+			},
+			expectPoolVersionChanged: true,
+		},
+		{
+			name: "poolVersion changes with ImageSet",
+			existing: []runtime.Object{
+				initializedPoolBuilder.Build(testcp.WithImageSet("abc123")),
+			},
+			expectPoolVersionChanged: true,
+		},
+		{
+			name: "poolVersion changes with InstallConfigSecretTemplate",
+			existing: []runtime.Object{
+				initializedPoolBuilder.Build(testcp.WithInstallConfigSecretTemplateRef("abc123")),
+			},
+			expectPoolVersionChanged: true,
```

Comment:
this could be a smaller unit test just for the calculatePoolVersion function instead of putting it into this Reconcile test.

Reply:
i think those are separate, the reconcile test makes sure that some hash is set. while the unit test for calculatePoolVersion tests that all the necessary fields are included in hash changes.

Reply:
I just don't agree here. This controller does not have unit tests for individual helper functions. It does have cases that exercise them through the entire reconcile. The broader we can make a test for a given level of coverage, the better the test is. I'd like to leave this alone.

Reply:
Let's roll with it as is, I like separate unit tests for helpers of complexity but this the nature of this one doesn't scream separate to me and seems ok in a broader unit test.